megamek/src/megamek/client/ui/swing/TargetChoiceDialog.java

```diff
@@ -0,0 +1,159 @@
+import megamek.common.actions.WeaponAttackAction;
+import megamek.common.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.List;
+
+/**
+ * A modal dialog for choosing one or more Targetable objects. Can show stats
+ * in brief or in detail.
+ */
+public class TargetChoiceDialog extends AbstractChoiceDialog<Targetable> {
+    final ClientGUI clientGUI;
+    Entity firingEntity;
+
+     /**
+     * This creates a modal dialog to pick one or more Targetable objects.
+     * @param frame         parent @JFrame that owns this dialog
+     * @param title         Resource key string only, plain text will result in NPE from Resources
+     * @param message       HTML or plain text message show at top of dialog
+     * @param targets       things to chose from
+     * @param isMultiSelect if true, allows user to select multiple items. if false first,
+     *                      item chosen will close the window
+     * @param clientGUI     Needed to look up details of some targetables such as buildings
+     */
+    protected TargetChoiceDialog(JFrame frame, String title, String message,
+                                 @Nullable List<Targetable> targets, boolean isMultiSelect,
+                                 ClientGUI clientGUI) {
+        super(frame, title, message, targets, isMultiSelect);
+        this.clientGUI = clientGUI;
+        // initialize must be called after all member variables set
+        initialize();
+    }
+
+    /**
+     * This creates a modal dialog to pick one or more Targetable objects.
+     * @param frame         parent @JFrame that owns this dialog
+     * @param title         Resource key string only, plain text will result in NPE from Resources
+     * @param message       HTML or plain text message show at top of dialog
+     * @param targets       things to chose from
+     * @param isMultiSelect if true, allows user to select multiple items. if false first,
+     *                      item chosen will close the window
+     * @param clientGUI     Needed to look up details of some targetables such as buildings
+     * @param firingEntity Needed to determine ToHit modifiers for targets
+     */
+    protected TargetChoiceDialog(JFrame frame, String title, String message,
+                                 @Nullable List<Targetable> targets, boolean isMultiSelect,
+                                 ClientGUI clientGUI, @Nullable Entity firingEntity) {
+        super(frame, title, message, targets, isMultiSelect);
+        this.clientGUI = clientGUI;
+        this.firingEntity = firingEntity;
+        // initialize must be called after all member variables set
+        initialize();
+    }
+
+    @Override
+    protected void detailLabel(JToggleButton button, Targetable target) {
+        if (target instanceof Entity) {
+            button.setText("<html>" + infoText(target) + UnitToolTip.getEntityTipVitals((Entity) target, null) + "</html>");
+        } else if (target instanceof BuildingTarget) {
```

Comment:
I seem to recall that there's a TargetType property that covers this sort of thing; I could be wrong though.

Reply:
There is a Targetable.getTargetType() but it is more specific and maybe not useful for this code, which is just passing onto the Tooltip helpers to handle the details. 
For instance,  BuildingTarget can have several TargetTypes, including TYPE_BLDG_IGNITE, TYPE_BLDG_IGNITE, TYPE_BLDG_TAG. Does this code care a new one is added or removed or the flags get redone? I don't think it does, so keeping the instanceof is more maintainable. But, of course, will do as you guys think correct.

Reply:
Actually - I think maybe I should move this if statement into a static method in tooltips, as I have encountered another case where I want to handle this. Please hold off on review (sorry, can;t seem to turn this back into a draft)

Reply:
@NickAragua What do you think of the keeping the instanceof for maintainability? Like I said, either way is fine with me, just let me know.